src/garage/envs/multi_env_wrapper.py

```diff
@@ -145,101 +175,101 @@ def active_task_index(self):
             int: Index of active task.
 
         """
-        if hasattr(self.env, 'active_task_index'):
-            return self.env.active_task_index
+        if hasattr(self._env, 'active_task_index'):
+            return self._env.active_task_index
         else:
             return self._active_task_index
 
-        return self._active_task_index
-
-    @property
-    def observation_space(self):
-        """Observation space.
+    def reset(self):
+        """Sample new task and call reset on new task env.
 
         Returns:
-            akro.Box: Observation space.
+            numpy.ndarray: The first observation. It must conform to
+                `observation_space`.
+            dict: The episode-level information. Note that this is not part
+                of `env_info` provided in `step()`. It contains information of
+                the entire episode， which could be needed to determine the
+                first action (e.g. in the case of goal-conditioned or MTRL.)
 
         """
+        self._active_task_index = self._sample_strategy(
+            self._num_tasks, self._active_task_index)
+        self._env = self._task_envs[self._active_task_index]
+        obs, episode_info = self._env.reset()
+        self._step_cnt = 0
         if self._mode == 'vanilla':
-            return self.env.observation_space
+            pass
         elif self._mode == 'add-onehot':
-            task_lb, task_ub = self.task_space.bounds
-            env_lb, env_ub = self._observation_space.bounds
-            return akro.Box(np.concatenate([env_lb, task_lb]),
-                            np.concatenate([env_ub, task_ub]))
+            obs = np.concatenate([obs, self._active_task_one_hot()])
         else:  # self._mode == 'del-onehot'
-            env_lb, env_ub = self._observation_space.bounds
-            num_tasks = self._num_tasks
-            return akro.Box(env_lb[:-num_tasks], env_ub[:-num_tasks])
+            obs = obs[:-self._num_tasks]
 
-    @observation_space.setter
-    def observation_space(self, observation_space):
-        """Observation space setter.
+        self._last_observation = obs
 
-        Args:
-            observation_space (akro.Box): Observation space.
+        return obs, episode_info
 
-        """
-        self._observation_space = observation_space
+    def step(self, action):
+        """Step the active task env.
 
-    def _active_task_one_hot(self):
-        """One-hot representation of active task.
+        Args:
+            action (object): object to be passed in Environment.reset(action)
```

Comment:
This docstring doesn't make any sense to me -- reset doesn't have an argument, and action should be an `np.ndarray`